### ISSUE TITLE
feat: migrate from Expo Router to React Navigation

### ISSUE THREAD

**Issue by grabbou** [rise-tools/rise-tools#112]
This is an ongoing umbrella issues. Features or bugs will be added on an ongoing basis to track importance:

- Not being able to choose screen presentation - there is no filesystem and/or programatic API to do so

**Comment by ericvicenti** [comment]
screen presentation, you want to use modals? I guess you looked into this: https://docs.expo.dev/router/advanced/modals/

for defining screens ahead of time I think it can be done with expo router. the tricky part is that we don't have the metadata like title handy at the time when we push to that route. if we did, there are ways

**Comment by grabbou** [comment]
Initially, I somewhat thought this was still available in React Navigation:
```
router.push(route, { presentation: 'modal' })
```
but I think it's gone for a while now! 😆 

Apparently, it is similar in both routers, so I will update the issue to focus on screen map instead.

**Comment by grabbou** [comment]
Yeah, it kinda sucks but I think it could be possible with Expo Router to do the following:

```
import { Stack } from 'expo-router';
export default function Layout() {
  return (
    <Stack>
     {screens.map(screen => (
      <Stack.Screen
        name={screen.name}
        options={screen.options}
      />
      )}
    </Stack>
  );
}
```

However, [as per their documentation](https://docs.expo.dev/router/advanced/modals/), it appears all screens need to have a corresponding file on the filesystem, which is not the case with React Navigation. So, unless I am wrong, we would have to move away from Expo Router or find an API to register a screen programatically (there's one for sure, question whether it's exported).

To make this work, we need to solve:

> for defining screens ahead of time I think it can be done with expo router. the tricky part is that we don't have the metadata like title handy at the time when we push to that route. if we did, there are ways

Do you have any thoughts? 

I have an idea (maybe stupid, but worth checking if possible): In your default model, e.g. "inventory", define "Stack" and "Stack Screen" however you want.

```tsx
import { Stack } from '@rise-tools/kit-expo-router';
export const models = {
   inventory: Inventory,
   inventoryItem: InventoryItem,
}

export function InventoryRouter() {
  return (
    <Stack>
      <Stack.Screen
          name="inventory"
          options={{ title: "Inventory "}}
      />
     <Stack.Screen
          name="inventoryItem"
          options={{ title: "Inventory Item"}}
      />
    </Stack>
  );
}
```

Even better, we could further leverage that to get rid of models map, e.g.:
```tsx
import { Stack } from '@rise-tools/kit-expo-router';

const Inventory = view(() => {}) // view state model
const InventoryItem = () => { // fn state model

}

export function InventoryRouter() {
  return (
    <Stack>
      <Stack.Screen
          name="home"
          options={{ title: "Inventory "}}
          modelState={Inventory} // React Navigation has "component", we have "modelState"
      />
     <Stack.Screen
          name="item"
          options={{ title: "Inventory Item"}}
          modelState={InventoryItem} // React Navigation has "component", we have "modelState"
      />
    </Stack>
  );
}
``` 

and in the main file:
```ts
import { Inventory } from './inventory'
createWSServer([inventory], Number(process.env.PORT || '3005'))
```

and for each item in the array, call it to get JSON and convert it to:
```
{ "inventory": InventoryRouter, "inventory/home": Inventory, "inventory/item": inventoryItem }
```
to make it all available as valid routes.

On top of that, return the component that is currently being accessed. Typically, the user will first call "inventory" which will return  configuration pointing to the default route and then, subsequent request will be fired to get data for that path. Can probably be improved, but just an idea.
